forge/ee/lib/billing/index.js

```diff
@@ -531,6 +531,57 @@ module.exports.init = async function (app) {
             subscription.trialEndsAt = null
             subscription.trialStatus = app.db.models.Subscription.TRIAL_STATUS.ENDED
             await subscription.save()
+        },
+
+        /**
+         * Resynchronises a subscription with stripe. This ensures are local view
+         * of the subscription status matches stripe's view.
+         */
+        resyncTeamSubscription: async (team, subscription) => {
+            if (subscription.subscription) {
+                try {
+                    const stripeSubscription = await stripe.subscriptions.retrieve(subscription.subscription)
+                    if (stripeSubscription.status !== subscription.status) {
```

Comment:
I think we should always log here and perhaps add a guard. e.g. if stripeSubscription is null

Reply:
Guard is not strictly necessary as it throws an err if the subscription isn't found - so stripeSubscription should never be null.